File: app/src/main/java/com/bq/zowi/interactors/CheckInstalledZowiAppInteractor.java
package com.bq.zowi.interactors;

import android.support.annotation.Nullable;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface CheckInstalledZowiAppInteractor {
    public static final String MAIN_ZOWI_APP_ID_PREFFIX = "ZOWI_BASE_v";

    public enum InstalledZowiAppStatus {
        UNKNOWN,
        OK,
        UPDATE_NEEDED,
        CUSTOM_FIRMWARE
    }

    Single<InstalledZowiAppStatus> checkAndManageInstalledZowiApp();

    InstalledZowiAppStatus manageInstalledZowiApp(@Nullable String str);
}

File: zowi-core/src/main/kotlin/com/bq/zowi/usecases/CheckInstalledZowiAppInteractor.kt
package com.bq.zowi.usecases

import io.reactivex.Single

interface CheckInstalledZowiAppInteractor {
    companion object {
        const val MAIN_ZOWI_APP_ID_PREFFIX = "ZOWI_BASE_v"
    }

    enum class InstalledZowiAppStatus {
        UNKNOWN, OK, UPDATE_NEEDED, CUSTOM_FIRMWARE
    }

    fun checkAndManageInstalledZowiApp(): Single<InstalledZowiAppStatus>
    fun manageInstalledZowiApp(installedZowiAppId: String?): InstalledZowiAppStatus
}
